holoviews/operation/datashader.py

```diff
@@ -1725,3 +1725,193 @@ class directly_connect_edges(_connect_edges, connect_edges):
+      Function that transforms the hits dataframe after applying the
+      transformer and before it it pass to the Points. Can be used to
+      customize the value dimensions e.g to implement custom hover
+      behavior.""")
+
+    # Stream values and overrides
+    streams = param.List(default=[PointerXY])
+    x = param.Number(default=0)
+    y = param.Number(default=0)
+
+    def _process(self, raster, key=None):
+        if isinstance(raster, RGB):
+            raster = raster[..., raster.vdims[-1]]
+        x_range, y_range = raster.range(0), raster.range(1)
+        xdelta, ydelta = self._distance_args(raster, x_range, y_range, self.p.pixels)
+        x, y = self.p.x, self.p.y
+        val = raster[x-xdelta:x+xdelta, y-ydelta:y+ydelta].reduce(function=np.nansum)
+        if np.isnan(val):
+            val = self.p.null_value
+
+        if ((self.p.value_bounds and
+             not (self.p.value_bounds[0] < val < self.p.value_bounds[1]))
+             or val == self.p.null_value):
+            empty_df = self._empty_df(raster.dataset)
+            df = self.p.points_transformer(self.p.transformer(empty_df))
+            self.hits = self.p.hits_transformer(self.p.transformer(empty_df))
+            vdims = self._vdims(raster, df)
+            return Points(df, kdims=raster.kdims, vdims=vdims)
+
+        masked = self._mask_dataframe(raster, x, y, xdelta, ydelta)
+        dist_sorted = self._sort_by_distance(raster, masked, x, y)
+        df = self.p.points_transformer(self.p.transformer(dist_sorted))
+        self.hits = self.p.hits_transformer(self.p.transformer(dist_sorted))
+        vdims = self._vdims(raster, df)
+        return Points(df, kdims=raster.kdims, vdims=vdims).iloc[:self.p.point_count]
+
+    @property
+    def mask(self):
+        return inspect_mask.instance(pixels=self.p.pixels)
+
+    @classmethod
+    def _vdims(cls, raster, df):
+        ds = raster.dataset
+        if 'spatialpandas' in ds.interface.datatype:
+            coords = [ds.interface.geo_column(ds.data)]
+        else:
+            coords = [kd.name for kd in raster.kdims]
+        return [col for col in df.columns if col not in coords]
+
+    @classmethod
+    def _empty_df(cls, dataset):
+        if 'dask' in dataset.interface.datatype:
+            return dataset.data._meta.iloc[:0]
+        elif dataset.interface.datatype in ['pandas', 'geopandas', 'spatialpandas']:
+            return dataset.data.head(0)
+        return dataset.iloc[:0].dframe()
+
+    @classmethod
+    def _distance_args(cls, element, x_range, y_range,  pixels):
+        ycount, xcount =  element.interface.shape(element, gridded=True)
+        x_delta = abs(x_range[1] - x_range[0]) / xcount
+        y_delta = abs(y_range[1] - y_range[0]) / ycount
+        return (x_delta*pixels, y_delta*pixels)
+
+    @classmethod
+    def _mask_dataframe(cls, raster, x, y, xdelta, ydelta):
+        "Mask the dataframe with size around x and y"
```

Comment:
What is `size`?

Reply:
The size parameter (in pixels). I'll clarify accordingly.

Reply:
Addressed in efa5fff18